frontend/src/components/widgets/FileUploader/FileUploader.tsx

```diff
@@ -129,135 +177,131 @@ class FileUploader extends React.PureComponent<Props, State> {
       })
   }
 
-  private reset = (): void => {
-    this.setState({
-      status: "READY",
-      errorMessage: undefined,
-      acceptedFiles: [],
+  private rejectFiles = (rejectedFiles: FileRejection[]): void => {
+    rejectedFiles.forEach((rejectedFile, index) => {
+      Object.assign(rejectedFile.file, {
+        status: FileStatuses.ERROR,
+        errorMessage: this.getErrorMessage(
+          rejectedFile.errors[0].code,
+          rejectedFile.file
+        ),
+      })
+      this.handleFile(rejectedFile.file, index)
```

Comment:
So actually reverting this to be mutable. In order to make this immutable in the current state, we'd have to clone the file and since the file could be a large file, I don't think it'd be great to create new copies of the file as we process it. 

I can create a wrapper around the file and pass these attributes into the wrapper rather than the file itself. However, this would be something I prefer to take on in a separate PR and probably as a follow up post launch (#2094)